app/v2/controllers/V2MovementsController.scala

```diff
@@ -228,20 +240,75 @@ class V2MovementsControllerImpl @Inject() (
     }
 
   def getMessage(movementType: MovementType, movementId: MovementId, messageId: MessageId): Action[AnyContent] =
-    (authActionNewEnrolmentOnly andThen acceptHeaderActionProvider(acceptOnlyJson = false)).async {
+    (authActionNewEnrolmentOnly andThen acceptHeaderActionProvider(jsonAndJsonWrappedXmlAcceptHeaders)).async {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+        (for {
+          messageSummary <- persistenceService.getMessage(request.eoriNumber, movementType, movementId, messageId).asPresentation
+          acceptHeader = request.headers.get(HeaderNames.ACCEPT).get
+          body <-
+            if (messageSummary.uri.isDefined) processLargeMessage(movementId, movementType, messageSummary, acceptHeader)
```

Comment:
In that case we would have a really big set of case statements to separate the logic for small and large so I'd rather keep the implementation in separate functions but I've made the code more readable/functional by using a match statement instead of an if statement